saiph/reduction/famd_test.py

```diff
@@ -209,6 +209,51 @@ def test_get_variable_contributions(mixed_df: pd.DataFrame) -> None:
     assert_frame_equal(cos2, expected_cos2, check_exact=False, atol=0.0001)
 
 
+def test_get_variable_contributions_with_model_fitted_on_other_df() -> None:
+    """Verify that the contributions and cos2 are the ones expected."""
+    df_to_fit_model = pd.DataFrame(
+        {
+            "variable_1": [1, 2, 3, 4],
+            "tool": ["hammer", "toaster", "hammer", "toaster"],
+        }
+    )
+    df_to_compute_contributions = pd.DataFrame(
+        {
+            "variable_1": [2, 2, 3, 1],
+            "tool": ["toaster", "toaster", "hammer", "hammer"],
+        }
+    )
+    _, model = fit_transform(df_to_fit_model, nf=3)
+
+    contributions, cos2 = get_variable_contributions(
+        model, df_to_compute_contributions, explode=False
+    )
```

Comment:
how is it possible get the contribution of a df that was not used to build the model? IMO this should be impossible, what do you think?

Reply:
It is not possible you are right ! That is why I couldn't find a better solution than fitting on the new set